src/lib/comms/sol-oic-security.c

```diff
@@ -53,34 +51,51 @@ struct cred_item {
     /* FIXME: Only symmetric pairwise keys supported at the moment. */
 };
 
-struct creds {
-    struct sol_vector items;
-    const struct sol_oic_security *security;
-};
-
 struct sol_socket *sol_coap_server_get_socket(const struct sol_coap_server *server);
 
 static ssize_t
 creds_get_psk(const void *data, struct sol_str_slice id,
     char *psk, size_t psk_len)
 {
-    const struct creds *creds = data;
-    struct cred_item *iter;
-    uint16_t idx;
+    SOL_BUFFER_DECLARE_STATIC(buf, PATH_MAX);
```

Comment:
instead of `PATH_MAX`, use `NAME_MAX` since this is just the base name, not the full path. It will help keep stack smaller and remember this will be enabled on the small OSes as well.